pkg/protoutil/paths/paths.go

```diff
@@ -0,0 +1,306 @@
+package paths
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protopath"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+func Parse(root protoreflect.MessageDescriptor, pathStr string) (protopath.Path, error) {
```

Comment:
I think I understand some of this... I was surprised that there's a `protopath` package but this parsing/derefencing functionality doesn't exist? Weird.

There might be some gaps in this code, but the unit tests are good.

Reply:
Yeah, the builtin protopath is rather incomplete. I've written a number of variations of this function for so many projects, maybe I should just try to contribute it to upstream instead, or collect all this code into a library